integration/truncindex_test.go

```diff
@@ -0,0 +1,135 @@
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+func genTruncIndex(normalName string) string {
+	return normalName[:(len(normalName)+1)/2]
+}
+
+func TestTruncIndex(t *testing.T) {
+	t.Logf("Pull an image")
+	const appImage = "busybox"
+	imgID, err := imageService.PullImage(&runtimeapi.ImageSpec{Image: appImage}, nil)
+	assert.NoError(t, err)
+	imgTruncId := genTruncIndex(imgID)
+	defer func() {
+		assert.NoError(t, imageService.RemoveImage(&runtimeapi.ImageSpec{Image: imgTruncId}))
+	}()
+
+	t.Logf("Get image status by truncindex, truncId: %s", imgTruncId)
+	res, err := imageService.ImageStatus(&runtimeapi.ImageSpec{Image: imgTruncId})
+	assert.NoError(t, err)
+	assert.NotEqual(t, nil, res)
+	assert.Equal(t, imgID, res.Id)
+
+	// TODO(yanxuean): for failure test case where there are two images with the same truncindex.
+	// if you add n images at least two will share the same leading digit.
+	// "sha256:n" where n is the a number from 0-9 where two images have the same trunc,
+	// for example sha256:9
+	// https://github.com/kubernetes-incubator/cri-containerd/pull/352
+	// I am thinking how I get the two image which have same trunc.
+
+	// TODO(yanxuean): add test case for ListImages
+
+	t.Logf("Create a sandbox")
+	sbConfig := PodSandboxConfig("sandbox", "truncindex")
+	sb, err := runtimeService.RunPodSandbox(sbConfig)
+	assert.NoError(t, err)
+	sbTruncIndex := genTruncIndex(sb)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sbTruncIndex))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sbTruncIndex))
+	}()
+
+	t.Logf("Get sandbox status by truncindex")
+	sbStatus, err := runtimeService.PodSandboxStatus(sbTruncIndex)
+	assert.NoError(t, err)
+	assert.Equal(t, sb, sbStatus.Id)
+
+	t.Logf("Forward port for sandbox by truncindex")
+	_, err = runtimeService.PortForward(&runtimeapi.PortForwardRequest{sbTruncIndex, []int32{80}})
+	assert.NoError(t, err)
+
+	// TODO(yanxuean): add test case for ListPodSandbox
+
+	t.Logf("Create a container")
+	cnConfig := ContainerConfig(
+		"containerTruncIndex",
+		appImage,
+		WithCommand("top"),
+	)
+	cn, err := runtimeService.CreateContainer(sbTruncIndex, cnConfig, sbConfig)
+	assert.NoError(t, err)
+	cnTruncIndex := genTruncIndex(cn)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cnTruncIndex))
+	}()
+
+	t.Logf("Get container status by truncindex")
+	cStatus, err := runtimeService.ContainerStatus(cnTruncIndex)
+	assert.NoError(t, err)
+	assert.Equal(t, cn, cStatus.Id)
+
+	t.Logf("Start the container")
+	assert.NoError(t, runtimeService.StartContainer(cnTruncIndex))
+	defer func() {
+		assert.NoError(t, runtimeService.StopContainer(cnTruncIndex, 10))
+	}()
+
+	t.Logf("Stats the container")
+	cStats, err := runtimeService.ContainerStats(cnTruncIndex)
+	assert.NoError(t, err)
```

Comment:
e.g. you may want to use `require` here to avoid panic.

Reply:
done